release-notes/swan-lake-2201.1.0-release-note.md

```diff
@@ -30,9 +30,364 @@ If you have not installed Ballerina, then download the [installers](/downloads/#
+
+```ballerina
+import ballerina/io;
+
+public function main() {
+    io:println(5.55.round(1)); // 5.6
+    decimal x = 5.55;
+    io:println(decimal:round(x)); // 6
+    io:println(decimal:round(5.55, fractionDigits = 0)); // 6
+    io:println(decimal:round(5.5565, fractionDigits = 3)); // 5.556
+}
+```
+
+##### An unreachable panic statement no longer results in a compilation error
+
+```ballerina
+function fn() returns string {
+    int|string a = 10;
+
+    if a is int {
+        return "INT";
+    } else {
+        return "STRING";
+    }
+
+    panic error("Not Reached!"); // unreachable, but not an error.
+}
+```
+
+##### `error:Clonable` Cloneable definition in error module is now public
+
+```ballerina
+import ballerina/io;
+
+public function main() {
+    error:Cloneable x = 4;
+    error e1 = error("reason 1", message = "My Detail");
+    map<error:Cloneable> m1 = e1.detail();
+
+    io:println(x);  // 4
+    io:println(m1); // {"message":"My Detail"}
+}
+```
+
+#### Bug Fixes
+
+- An invalid sub-typing relationship between `table` and `anydata` has been fixed.
+
+```ballerina
+public function main() {
+    table<map<any>> tany = table [{"a": 2}];
+    anydata ad = tany; // Results in a compilation error.
+}
+```
+
+- A union containing the `null` literal allowing `"null"` as a valid value has been fixed.
+
+```ballerina
+type Foo boolean|null;
+
+public function main() {
+    Foo a = "null"; // compilation error
+    "string"|null b = "null"; // compilation error
+}
+```
+
+- An issue that caused the value of enum members defined with quoted identifiers to include the quote has been fixed.
+
+```ballerina
+import ballerina/io;
+
+public enum Status {
+    'new,
+    old
+}
+public function main() {
+    io:println('new); // Previously printed `'new`, now prints `new`.
+}
+```
+
+- A compilation error is now logged when there is an extra comma inside a mapping match pattern.
+
+```ballerina
+type MyRecord record {
+    int field1;
+    int field2;
+};
+
+function fn(MyRecord r1) {
+    match r1 {
+        {field1: 0,} => { // A syntax error is now given for the comma.
+
+        }
+    }
+}
+```
+
+- An issue in the parser that disallowed qualified identifiers with a pre-declared prefix as a const match pattern has been fixed.
+
+```ballerina
+function fn(any x) {
+    match x {
+        int:MAX_VALUE => { // Match pattern is now allowed
+        }
+
+        [int:MAX_VALUE, int:MIN_VALUE] => { // Match pattern is now allowed
+        }
+    }
+}
+```
+
+- Qualified identifiers not being allowed in error match patterns has been fixed.
+
+```ballerina
+function fn(error e) {
+    match e {
+        error(errors:MESSAGE) => { // Match pattern is now allowed
+
+        }
+    }
+}
+```
+
+- The inherent type of a list constructed using a list constructor with `any` as the contextually-expected type has been fixed to be `(any|error)[]` instead of `any[]`.
```

Comment:
```suggestion
- Fixed the inherent type of a list constructed using a list constructor with `any` as the contextually-expected type to be `(any|error)[]` instead of `any[]`
```